src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js

```diff
@@ -0,0 +1,317 @@
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const switchToSignupButton = componentWrapper.find("#switchToSignupButton");
+    expect(switchToSignupButton.exists()).toBe(true);
+    await switchToSignupButton.trigger("click");
+  });
+
+  it("Check if the small titles work", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const smallTitles = componentWrapper.find("#small");
+    expect(smallTitles.exists()).toBe(true);
+  });
+
+  it("Check if the big titles work", async () => {
+    const componentWrapper = mount(Login, {
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const bigTitles = componentWrapper.find("#big");
+    expect(bigTitles.exists()).toBe(true);
+  });
+});
+
+describe("Signup.vue", () => {
+  it("renders with null props", () => {
+    const componentWrapper = mount(Signup, {
+      props: {
+        /*
+          TODO(Need to add API tests here)
+        */
+      },
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    // exists
+    expect(componentWrapper.exists()).toBe(true);
+  });
+
+  // The following are the component tests, all passing
+  it("Check if the signup_components_container exists", async () => {
+    const componentWrapper = mount(Signup, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const signup_components_container = componentWrapper.find(
+      ".signup_components_container"
+    );
+    expect(signup_components_container.exists()).toBe(true);
+  });
+
+  it("Check if the signupForm works", async () => {
+    const componentWrapper = mount(Signup, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const signupForm = componentWrapper.find("#signupForm");
+    expect(signupForm.exists()).toBe(true);
+    await signupForm.trigger("submit.prevent");
+  });
+
+  it("Check if the switchToLoginButton works", async () => {
+    const componentWrapper = mount(Signup, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const switchToLoginButton = componentWrapper.find("#switchToLoginButton");
+    expect(switchToLoginButton.exists()).toBe(true);
+    await switchToLoginButton.trigger("click");
+  });
+
+  it("Check if the signupThroughSSO works", async () => {
+    const componentWrapper = mount(Signup, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const signupWithGoogle = componentWrapper.find("#signupWithGoogle");
+    expect(signupWithGoogle.exists()).toBe(true);
+    await signupWithGoogle.trigger("submit.prevent");
```

Comment:
Add what to expect after trigger please

Reply:
What would you do in this case, as you have experience successfully writing tests? These are the exact guidance I needed.